hotelservice/Services/Hotel/HotelService.cs

```diff
@@ -87,6 +87,36 @@ public async Task<GetHotelResponse> GetHotel(GetHotelRequest request)
 
     public async Task<HotelBookRoomsResponse> BookRooms(HotelBookRoomsRequest request)
     {
+        
+        var hotelId = request.BookingDetails.Id;
+
+        var popularHotel = await _dbContext.PopularHotels
+            .FirstOrDefaultAsync(h => h.Id == hotelId);
+        
+        var popularHotelInfo = await _dbContext.Hotels
```

Comment:
This variable is redundant - the same thing is loaded and checked just below added code

Reply:
Fixed, used fetched hotel instead.